torch/csrc/instruction_counter/Module.cpp

```
@@ -0,0 +1,87 @@
+#include <torch/csrc/instruction_counter/Module.h>
+#include <torch/csrc/utils/pybind.h>
+#include <cerrno>
+#include <cstdint>
+#include <cstdio>
+#include <cstdlib>
+#include <cstring>
+
+#if defined(__linux__)
+#include <linux/perf_event.h>
+#include <sys/ioctl.h>
+#include <sys/syscall.h>
+#include <unistd.h>
+#endif
+
+namespace torch::instruction_counter {
+
+long start() {
+#if !defined(__linux__)
+  printf("This systems seems not to be Linux");
+  return -1;
```

Raise an exception instead of returning -1?